src/azure-cli/azure/cli/command_modules/vm/_params.py

```diff
@@ -458,6 +458,9 @@ def load_arguments(self, _):
         c.argument('port', help="The port or port range (ex: 80-100) to open inbound traffic to. Use '*' to allow traffic to all ports. Use comma separated values to specify more than one port or port range.")
         c.argument('priority', help='Rule priority, between 100 (highest priority) and 4096 (lowest priority). Must be unique for each rule in the collection.', type=int)
 
+    with self.argument_context('vm list') as c:
+        c.argument('vmss', min_api='2021-11-01', help='The system query option to filter VMs. Allowed value is vmss id.')
```

Comment:
May I ask if it supports the name of VMSS?

Reply:
The API only accepts the full resource ID. I think there are two options:

1) Could CLI infer the vmss id from the parameters? 
`az vm list --vmss myvmss --resource-group demorg`

- That seems like the most customer friendly
- I think you would have to require the resource group if customer specifies `--vmss`. 
- You'd have to choose to make that call either at the subscription level or the resource group level (you can pass the filter to either)

2) Just make the parameter `--vmss-id`

- Simpler
- Doesn't seem as user friendly though

Reply:
OK, thanks for your good suggestion~